### ISSUE TITLE
MSID: 92080 Version: 1 DOI: arXiv:2209.08066v2

### ISSUE THREAD

**Issue by nlisgo** [elifesciences/enhanced-preprints-import#1863]
[PLACE MANUSCRIPT AND EDITOR DETAILS HERE WHEN AVAILABLE] See step 3

[PLACE PDF URL HERE WHEN AVAILABLE] See step 7

## Step 1. Inform bioRxiv

Who can help: @QueenKraken, @nlisgo, @scottaubrey

- [ ] Manuscript is in data hub index (https://data-hub-api.elifesciences.org/enhanced-preprints/docmaps/v2/by-publisher/elife/get-by-manuscript-id?manuscript_id=92080)

or (only one should be ticked. remove other from description.)

- [ ] Confirmed with Ted at bioRxiv that MECA archive is available

Send the following email to Ted and wait for his reply.

```
Hi Ted,

Please can you prepare the preprint MECA for 10.48550/arXiv:2209.08066v2

Thanks
```

## Step 2. Create preview of manuscript

Who can help: @fred-atherden, @nlisgo, @scottaubrey

- [ ] Manuscript is available for preview (https://prod--epp.elifesciences.org/preview/92080/v1)
    - [ ] Trigger import for staging: https://prod--epp.elifesciences.org/import
    - [ ] Sync data folder with s3://prod-elife-epp-data/data
    - [ ] Trigger import for staging: https://staging--epp.elifesciences.org/import
    - [ ] Sync data folder with s3://staging-elife-epp-data/data
    - [ ] Pull request created to [enhanced-preprint-data](https://github.com/elifesciences/enhanced-preprints-data/pulls)
    - [ ] Post link to pull request below

Pull request: [PLACE LINK TO PULL REQUEST HERE]

Detailed instructions: https://github.com/elifesciences/enhanced-preprints-data#add-a-manuscript

## Step 3: Awaiting public reviews and QC

Who can help: Production team

- [ ] QC passed or QC not needed

Who can help: Editorial team

- [ ] Public reviews available on sciety (https://sciety.org/articles/activity/10.48550/arXiv:2209.08066v2)
- [ ] Manuscript and editor details at the top of this issue (Supplied by Editorial team)
- [ ] Reviews returned by EPP - https://prod--epp.elifesciences.org/api/reviewed-preprints/92080/v1/reviews

<details>
<summary>Example</summary>

```
"msas": "Genetics and Genomics", "Neuroscience"
"msid": "84628"
"version": "1"
"preprintDoi": "10.1101/2022.10.28.514241"
"articleType": "Reviewed Preprint"
"status": "Published from the original preprint after peer review and assessment by eLife."

"Reviewed Preprint posted": "2023-01-02"
"Sent for peer review": "2022-10-28"
"Posted to bioRxiv": "2022-11-21" (link: "Go to bioRxiv": "https://www.biorxiv.org/content/10.1101/2022.10.28.514241v1")

Editors:
Reviewing Editor
Michael B Eisen
University of California, Berkeley, United States
Senior Editor
Michael B Eisen
University of California, Berkeley, United States
```

</details>

## Step 4: Modify manuscripts.json (no PDF)

- [ ] Reviewed preprint is published to EPP (https://elifesciences.org/reviewed-preprints/92080)
    - [ ] Pull request created to [enhanced-preprint-client](https://github.com/elifesciences/enhanced-preprints-client/pulls)
    - [ ] Pull request switched to "Ready for review"
    - [ ] Post link to pull request below
- [ ] Request that a doi be registered
    - [ ] Post link to [#enhanced-preprint](https://elifesciences.slack.com/archives/C03EVJSUA77) comment below

Pull request: [PLACE LINK TO PULL REQUEST HERE]
[#enhanced-preprint](https://elifesciences.slack.com/archives/C03EVJSUA77) comment thread: [PLACE LINK TO COMMENT HERE]

<details>
<summary>Instructions to modify manuscripts.json</summary>

- Visit: https://github.com/elifesciences/enhanced-preprints-client/actions/workflows/publish-manuscript.yaml
- Click: Run workflow
- Complete the form and click "Run workflow"
- A successful run should result in a new pull request at https://github.com/elifesciences/enhanced-preprints-client/pulls
- Open the pull request and click the "Ready for review" button to trigger tests
- Once the tests pass and you are happy with the changes the PR can be merged

Example pull request: https://github.com/elifesciences/enhanced-preprints-client/pull/334/files

Once the pull request is merged in it should be available a few minutes later.

</details>

### Request that a doi

Post the following in [#enhanced-preprint](https://elifesciences.slack.com/archives/C03EVJSUA77):

```
@Fred can you register a doi for https://elifesciences.org/reviewed-preprints/92080
```

## Step 5: Awaiting search reindex

- [ ] Reviewed preprint is avaliable on journal homepage (https://elifesciences.org)

The search reindex is triggered once an hour. We need the reviewed preprint to be indexed as the search application serves the journal homepage.

<details>
<summary>Additional info</summary>

If needed, the jenkins pipeline to reindex search can be triggered sooner.

https://alfred.elifesciences.org/job/process/job/process-reindex-reviewed-preprints/

</details>

## Step 6: Published! Request PDF generation

- [ ] Post the link to the PDF url at the top of the issue
    - [ ] Post link to [#enhanced-preprint](https://elifesciences.slack.com/archives/C03EVJSUA77) comment below

[#enhanced-preprint](https://elifesciences.slack.com/archives/C03EVJSUA77) comment thread: [PLACE LINK TO COMMENT HERE]

Post the following to the [#enhanced-preprint](https://elifesciences.slack.com/archives/C03EVJSUA77) on slack:

```
@Ryan Dix-Peek please can you generate a PDF for https://elifesciences.org/reviewed-preprints/92080
```

## Step 7: Introduce PDF to data folder and git repo

- [ ] PDF is avaliable at https://github.com/elifesciences/enhanced-preprints-data/raw/master/data/92080/v1/92080-v1.pdf

Detailed instructions: https://github.com/elifesciences/enhanced-preprints-data#add-a-pdf

## Step 8: Add PDF url to manuscripts.json

- [ ] Reviewed preprint PDF is available for download (https://prod--epp.elifesciences.org/reviewed-preprints/92080)
    - [ ] Pull request created to [enhanced-preprint-client](https://github.com/elifesciences/enhanced-preprints-client/pulls)
    - [ ] Pull request switched to "Ready for review"
    - [ ] Post link to pull request below

[PLACE LINK TO PULL REQUEST HERE]

<details>
<summary>Instructions to add PDF url to manuscripts.json</summary>

- Visit: https://github.com/elifesciences/enhanced-preprints-client/actions/workflows/add-pdf-url-to-manuscript.yaml
- Click: Run workflow
- Complete the form and click "Run workflow"
- A successful run should result in a new pull request at https://github.com/elifesciences/enhanced-preprints-client/pulls
- Open the pull request and click the "Ready for review" button to trigger tests
- Once the tests pass and you are happy with the changes the PR can be merged

Example pull request: https://github.com/elifesciences/enhanced-preprints-client/pull/397/files

Once the pull request is merged in it should be available a few minutes later.

</details>

## Step 9: Done!

- [ ] Kettle is on!

**Comment by naushinthomson** [comment]
Reviews unavailable

**Comment by fred-atherden** [comment]
@HazalCiplak do you know why only the author response is included in [the DocMap](https://data-hub-api.elifesciences.org/enhanced-preprints/docmaps/v2/by-publisher/elife/get-by-manuscript-id?manuscript_id=92080):
<details>

<summary>DocMap</summary>

```json
{
    "@context": "https://w3id.org/docmaps/context.jsonld",
    "type": "docmap",
    "id": "https://data-hub-api.elifesciences.org/enhanced-preprints/docmaps/v2/by-publisher/elife/get-by-manuscript-id?manuscript_id=92080",
    "created": "2023-08-23T10:03:02+00:00",
    "updated": "2023-08-23T10:03:02+00:00",
    "publisher": {
        "account": {
            "id": "https://sciety.org/groups/elife",
            "service": "https://sciety.org"
        },
        "homepage": "https://elifesciences.org/",
        "id": "https://elifesciences.org/",
        "logo": "https://sciety.org/static/groups/elife--b560187e-f2fb-4ff9-a861-a204f3fc0fb0.png",
        "name": "eLife"
    },
    "first-step": "_:b0",
    "steps": {
        "_:b0": {
            "actions": [
                {
                    "participants": [],
                    "outputs": [
                        {
                            "type": "preprint",
                            "identifier": "92080",
                            "doi": "10.7554/eLife.92080.1",
                            "versionIdentifier": "1",
                            "license": "http://creativecommons.org/licenses/by/4.0/"
                        }
                    ]
                }
            ],
            "assertions": [
                {
                    "item": {
                        "type": "preprint",
                        "doi": "10.48550/arXiv.2209.08066",
                        "versionIdentifier": "2"
                    },
                    "status": "under-review",
                    "happened": "2023-09-18T13:51:42+00:00"
                },
                {
                    "item": {
                        "type": "preprint",
                        "doi": "10.7554/eLife.92080.1",
                        "versionIdentifier": "1"
                    },
                    "status": "draft"
                }
            ],
            "inputs": [
                {
                    "type": "preprint",
                    "doi": "10.48550/arXiv.2209.08066",
                    "url": "https://arxiv.org/abs/2209.08066",
                    "versionIdentifier": "2",
                    "published": "2023-08-21",
                    "content": [
                        {
                            "type": "computer-file",
                            "url": "s3://prod-elife-epp-meca/92080-v1-meca.zip"
                        }
                    ]
                }
            ],
            "next-step": "_:b1"
        },
        "_:b1": {
            "actions": [
                {
                    "participants": [],
                    "outputs": [
                        {
                            "type": "reply",
                            "published": "2023-11-28T10:00:50.466957+00:00",
                            "doi": "10.7554/eLife.92080.1.sa0",
                            "license": "http://creativecommons.org/licenses/by/4.0/",
                            "url": "https://doi.org/10.7554/eLife.92080.1.sa0",
                            "content": [
                                {
                                    "type": "web-page",
                                    "url": "https://hypothes.is/a/Apr9CI3VEe6JSHOASn0nBQ"
                                },
                                {
                                    "type": "web-page",
                                    "url": "https://sciety.org/articles/activity/10.48550/arXiv.2209.08066#hypothesis:Apr9CI3VEe6JSHOASn0nBQ"
                                },
                                {
                                    "type": "web-page",
                                    "url": "https://sciety.org/evaluations/hypothesis:Apr9CI3VEe6JSHOASn0nBQ/content"
                                }
                            ]
                        }
                    ]
                }
            ],
            "assertions": [
                {
                    "item": {
                        "type": "preprint",
                        "doi": "10.48550/arXiv.2209.08066",
                        "versionIdentifier": "2"
                    },
                    "status": "peer-reviewed"
                }
            ],
            "inputs": [
                {
                    "type": "preprint",
                    "doi": "10.48550/arXiv.2209.08066",
                    "url": "https://arxiv.org/abs/2209.08066",
                    "versionIdentifier": "2"
                }
            ],
            "previous-step": "_:b0"
        }
    }
}
```

</details>

I can see the other hypothesis comments here https://hypothes.is/groups/q5X6RWJ6/elife:



![Image](https://github.com/elifesciences/enhanced-preprints-import/assets/43879983/a9ab0627-c393-4be0-a687-64c2381c3058)

**Comment by fred-atherden** [comment]
@acollings, @hazalCiplak has found the issue - the author response has been added to this URL https://arxiv.org/abs/2209.08066, and the rest of the hypothesis comments have been added to https://arxiv.org/abs/2209.08066v2 (the version URL ending in `v2`).

Please can you assist with deleting the the author response from the non-versioned URL (https://arxiv.org/abs/2209.08066) and readding it onto the versioned one (https://arxiv.org/abs/2209.08066v2) - the hypothesis ID (to delete) is `Apr9CI3VEe6JSHOASn0nBQ`.

**Comment by acollings** [comment]
@fred-atherden done!

**Comment by HazalCiplak** [comment]
Thank you, Andy! The DocMap has been updated. 
Just for clarification, should we interpret this as no "author response" for this version of the manuscript? There don't appear to be any new evaluations added in Hypothesis after one with the wrong was deleted.

**Comment by acollings** [comment]
Hi @HazalCiplak I did add an author response to https://arxiv.org/abs/2209.08066v2, so hopefully it's there?!

**Comment by fred-atherden** [comment]
I'm still not seeing the DocMap updated. It only shows one author response.

Whereas when I search in this view I'm not seeing any author response:


![Image](https://github.com/elifesciences/enhanced-preprints-import/assets/43879983/b144c14e-0d6f-4240-9b2b-84135b9b865e)


@HazalCiplak any idea what we should do in order to fix this one?

**Comment by HazalCiplak** [comment]
Interesting, thank you for letting me know @fred-atherden, I will check this.

**Comment by HazalCiplak** [comment]
Not seeing the author response in the dashboard is a separate issue (I am investigating it) but the reason we don't see evaluations in the DocMap is the new added author response uri looks like `https://arxiv.org/abs/2209.08066` without version still. 

<img width="1106" alt="Screenshot 2023-12-04 at 17 59 32" src="https://github.com/elifesciences/enhanced-preprints-import/assets/6259479/8d5076a1-134c-450b-826d-5ed0ee05d0b8">

**Comment by fred-atherden** [comment]
Many thanks Hazal. @acollings Is this something you could help us with please?

**Comment by HazalCiplak** [comment]
Meanwhile I have made changes on the dashboard filter to see all related data for hypothesis, hopefully this will be helpful to see the data better now.

**Comment by acollings** [comment]
Hi @fred-atherden and @HazalCiplak I'm not exactly sure if I did the right thing, but I've removed the author response from the un-versioned URL. Hopefully there is still an author response on the versioned one (https://arxiv.org/abs/2209.08066v2)?

**Comment by fred-atherden** [comment]
Thanks Andy - I'm not seeing any changes yet. @HazalCiplak could you advise if that's done the trick? Thanks

**Comment by HazalCiplak** [comment]
Author response appears in the dashboard. Is there something happening in Hypothesis that is removing the version from the URL for this case? The screenshot below shows related data, and it is still without the version.

<img width="1318" alt="Screenshot 2023-12-06 at 09 32 56" src="https://github.com/elifesciences/enhanced-preprints-import/assets/6259479/502ceb05-d51f-49ed-a55f-8ba3189eb271">

**Comment by acollings** [comment]
I'm not sure, @HazalCiplak. I definitely added it to the versioned URL. Please let me know if I should try again.

**Comment by acollings** [comment]
Thanks @HazalCiplak! We just screen shared, and will look forward to hearing if it worked!

**Comment by HazalCiplak** [comment]
AuthorReponse is without version again, there is bug there who/where can we report this issue to 🤔

![Image](https://github.com/elifesciences/enhanced-preprints-import/assets/6259479/2c18724a-08d2-490d-9427-ee3bfc2f5dc8)

**Comment by fred-atherden** [comment]
I'm not familiar with how the posting works - was the author response posted via Kotahi or via hypothesis directly?

I have no idea how responsive hypothesis are, but publication of this RP has already been delayed for 2 weeks due to this issue. Is there no way we can resolve this problem beyond raising the bug with a third party and hoping they fix the issue before the end of the year?

**Comment by acollings** [comment]
(Just for reference, the original version of the Author Response would have been posted using Kotahi, but we've deleted and re-tried using hypothes.is several times!)

**Comment by acollings** [comment]
Hi @fred-atherden and @HazalCiplak I've deleted/moved all the reviews, so hopefully we'll be good to go??

**Comment by HazalCiplak** [comment]
All looks good now on DocMaps. Thank you, Andy!

The issue is that all URIs are without version now, but we're able to match them because of the rank usage in the code. Let's see how it will be if there is a revision after these reviews. If the URL is still without version, there may be some issues, but let's wait until then. And meanwhile it would be good to report this issue to hypothesis so they can fix it.

**Comment by fred-atherden** [comment]
Many thanks both!

**Comment by de-code** [comment]
If the URIs aren't including the version, does that mean we didn't publish them via Kotahi?

**Comment by acollings** [comment]
From what I can see, they are on V2 (I posted them to https://arxiv.org/abs/2209.08066v2): the little blue 5 shows the number of annotations for that version. I didn't use Kotahi as the "republish" button has been unreliable in the past. If you'd like me to delete again and post using Kotahi, let me know :-)

![Screenshot 2023-12-12 at 17 20 17](https://github.com/elifesciences/enhanced-preprints-import/assets/43173300/23fd3e74-c853-40a7-af97-6714105f9b03)

**Comment by de-code** [comment]
I see, I didn't realise republishing via Kotahi was unreliable.

The hypothesis annotations do seem to not include the version as Hazal suggested, this is [visible in the API response for one of the annotations](https://hypothes.is/api/annotations/RYHLfpkBEe6tEBeXyOeupQ).

In our DocMaps we seem to be able to handle that now due to all annotations using the same URL.

But for any other consumer of the Hypothes.is annotation it would just link to the URL without version.
I am not sure whether there are side effects we wouldn't be happy with. e.g. I would expect the annotations to show for any version of the arXiv preprint?

Is it likely that we will publish reviews on a future version of the same arXiv preprint?
Will it be okay to republish the Hypothes.is annotation then, if needed?

---

BTW I think this may be more of an issue with arXiv than Hypothes.is.
For example for the page source for `https://arxiv.org/abs/2209.08066v2` reports its canonical URL as `https://arxiv.org/abs/2209.08066` (without version), via the following snippet:

```html
<link rel="canonical" href="/abs/2209.08066"/><meta property="og:type" content="website" />
```

For comparison, `https://www.biorxiv.org/content/10.1101/2023.01.03.522515v2` uses `https://www.biorxiv.org/content/10.1101/2023.01.03.522515v2` via:

```html
<link rel="canonical" href="[https://www.biorxiv.org/content/10.1101/2023.01.03.522515v2](view-source:https://www.biorxiv.org/content/10.1101/2023.01.03.522515v2)" />
```

**Comment by acollings** [comment]
Thanks @de-code! 

> I am not sure whether there are side effects we wouldn't be happy with. e.g. I would expect the annotations to show for any version of the arXiv preprint?

I don't think that's a big issue

> Is it likely that we will publish reviews on a future version of the same arXiv preprint?

Yes!

> Will it be okay to republish the Hypothes.is annotation then, if needed?

Of course! (I do wonder if the same thing will happen using Kotahi, so perhaps it's better to try sooner?? It'll only take a few minutes on Thursday, if anyone would like me to try)

**Comment by HazalCiplak** [comment]
If it won't take too much of your time, it would be good to try with Kotahi, as this (having url without version) could potentially be an issue when new reviews are posted in DocMaps, and we will need to handle it by then as well.

**Comment by de-code** [comment]
If we do encounter issues using Kotahi, then we could maybe use the [Hypothes.is API](https://h.readthedocs.io/en/latest/api-reference/v1/#tag/annotations/paths/~1annotations/post) to copy an annotation over.
(As a sort of one-off we could maybe do that in a Colab notebook. Would need authentication, e.g. via an API key)

**Comment by acollings** [comment]
Hi @HazalCiplak and @fred-atherden I've used Kotahi's republish button (with author response included). Let me know if there are any problems!

**Comment by HazalCiplak** [comment]
Thanks again @acollings! It looks great.

<img width="1383" alt="Screenshot 2023-12-14 at 17 12 55" src="https://github.com/elifesciences/enhanced-preprints-import/assets/6259479/0aab59ed-50c2-47d6-bd75-19b20221cdf3">
